packages/modular-scripts/src/rename.ts

```diff
@@ -0,0 +1,150 @@
+import { Project } from 'ts-morph';
+import actionPreflightCheck from './utils/actionPreflightCheck';
+import getModularRoot from './utils/getModularRoot';
+import getWorkspaceInfo, { WorkSpaceRecord } from './utils/getWorkspaceInfo';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import * as logger from './utils/logger';
+
+// Derive types from array; this is handy to iterate against it later
+const dependencyTypes = [
+  'dependencies',
+  'devDependencies',
+  'peerDependencies',
+] as const;
+
+type DependencyType = typeof dependencyTypes[number];
+
+type DependencyObject = {
+  [key in DependencyType]: Record<string, string>;
+};
+
+interface PackageJson extends Partial<DependencyObject> {
+  name: string;
+}
+
+async function rename(
+  oldPackageName: string,
+  newPackageName: string,
+): Promise<void> {
+  const workspace = Object.entries(await getWorkspaceInfo());
+
+  logger.debug(`Checking for existence of ${oldPackageName} in workspace.`);
+  const oldPackage = workspace.find(
+    ([packageName]) => packageName === oldPackageName,
+  )?.[1];
+  if (!oldPackage) {
+    throw new Error(`Package ${oldPackageName} not found.`);
+  }
+  logger.debug(`Checking for collision with ${newPackageName} in workspace.`);
+  if (workspace.find(([packageName]) => packageName === newPackageName)) {
+    throw new Error(`Package ${newPackageName} already exists.`);
+  }
+
+  // Rename the directory. Do it first because this fails if there's a collision and we don't want to clean up later
+  const newPackageLocation = path.join(
+    getModularRoot(),
+    path.join(oldPackage.location, '..'),
+    newPackageName,
+  );
+
+  logger.log(`Moving ${oldPackage.location} → ${newPackageLocation}`);
+
+  await fs.move(oldPackage.location, newPackageLocation);
+
+  // Change name in package.json
+  const newPackageJsonLocation = path.join(
+    newPackageLocation,
+    './package.json',
+  );
+
+  const packageJson = (await fs.readJson(
+    newPackageJsonLocation,
+  )) as PackageJson;
+
+  logger.log(`Changing package name ${packageJson.name} → ${newPackageName}`);
+  packageJson.name = newPackageName;
+
+  await fs.writeJson(newPackageJsonLocation, packageJson, { spaces: 2 });
+
+  // Search for packages that depend on the renamed package
+  type PackageDepInfo = { packageData: WorkSpaceRecord; json: PackageJson };
+
+  logger.log(`Searching for packages depending on ${oldPackageName}`);
+  const dependingPackages: PackageDepInfo[] = (
+    await Promise.all(
+      workspace
+        .filter(([packageName]) => packageName !== oldPackageName)
+        .map(async ([_, packageData]) => ({
+          packageData,
+          json: (await fs.readJson(
+            path.join(getModularRoot(), packageData.location, './package.json'),
+          )) as PackageJson,
+        })),
+    )
+  ).filter((pkgJsonInfo) =>
+    dependencyTypes.some(
+      (depField) => pkgJsonInfo.json[depField]?.[oldPackageName],
```

Comment:
What's the use case here? If a package is using another package in the same monorepo, they shouldn't need to specify it as a dependency in their package.json. The one case I can think of is if they want to specify an older version of the dependency, in which case, you wouldn't want to change it to the new package name. Maybe I'm missing a use case here.

Reply:
> If a package is using another package in the same monorepo, they shouldn't need to specify it as a dependency in their package.json

Don't they need to specify the dependency in their package.json to get the other package symlinked? I'm fairly new to yarn workspaces but that's what the documentation example says: https://classic.yarnpkg.com/lang/en/docs/workspaces/#toc-how-to-use-it (note that's not necessary to build `package-a`, if you don't need it outside the monorepo you just point `main` or `module` or `exports` to `src`)

There's always the case that `package-b` just imports from `package-a`'s source using a relative path (eg `import libA from "../package-a/src/index.js"`), but then what's the advantage of having workspaces (decoupling packages etc)?

Reply:
They don't need to specify the dependency to use the package immediately. Because we set `"workspaces": ["packages/**]` and we call "yarn" after we create a new package with `/packages`, yarn creates a symlink to the local package in your node_modules, which can be imported/required using its package name. 

But they would need to add it explicitly is if they wanted to externalize the dependency from the build, which case would be to published version that can be installed from the registry. But I can also see the value of doing it anyways just in case they do have the dependency there.

Reply:
Done, removed renaming of dependencies in `package.json`s (I was used to rushjs, where you have to specify internal dependencies. Good to know!)